### ISSUE TITLE
Unclear where to provide tlsOptions

### ISSUE THREAD

**Issue by eliykat** [ldapts/ldapts#161]
`tlsOptions` can be provided as an argument to the `Client` constructor or to the `client.startTLS` method. However, after some debugging, it seems that these are mutually exclusive:
* the `Client` constructor configures SSL over ldaps
* the `client.startTLS` method configures SSL with StartTLS

Providing `tlsOptions` to the `Client` constructor when you're intending to use StartTLS gave me the following error:

> Client network socket disconnected before secure TLS connection was established

... and it took a bit of trial and error to figure out what was going on.

I suggest that the README is updated to note this distinction more clearly. I can submit a PR for this if you agree. (Or if I've just misread it please let me know.)

Another fix could be to have the `Client` to detect which use of `tlsOptions` is intended (based on the protocol in the server URL?) and handle it accordingly. In that case it would only be provided in the constructor. However I'm not sure if there's any drawback to this kind of auto-detection logic.

PS. Thank you for your work on this library!

**Comment by nelson777** [comment]
I'm having problems to certify myself if I'm correctly connected to a secure AD. I was getting an UNABLE_TO_VERIFY_LEAF_SIGNATURE error exactly because of these confuse options. 
So just to clarify: if I use tlsOption in the client contructor, the connection is already secure by itself ? Is there a way to check if a connection is secure or not ? just connecting/binding to ldaps:// is enough ?

Because the problem is that I'm trying to write UnicodePwd;binary attribute, which AFAIK is only writable through a secure connection and I'm still not able to write it. So I'm considering the connection might not be secure as I just passed tlsOptions on the constructor and didn't call startTLS. But if I do, I get a OperationsError.

Is there a way to make sure the connection is secure ?

**Comment by eliykat** [comment]
This is still an issue. @jgeurts what do you think?

**Comment by jgeurts** [comment]
@eliykat Yep, agreed that it would be nice to have clarity around when to use each approach. How does [this update to the README look](https://github.com/ldapts/ldapts/pull/166)? Feel free to make any changes, corrections, etc!

**Comment by eliykat** [comment]
Looks good! I've left minor feedback on the PR. @nelson777 note that #166 also mentions your error.

**Comment by jgeurts** [comment]
Thanks for your help with this! The documentation is updated w/ the merging of #166
